### PR TITLE
feat: set challenge cookie data on lend filter

### PR DESCRIPTION
**Pull request by christian14b**

### CODE REVIEW

File: src/pages/Lend/LoanSearchPage.vue [hunk 1]
```diff
@@ -240,6 +241,7 @@ export default {
 				this.challengeData = goalsData?.goals?.values?.[0] || {};
 				const teamData = this.apollo.readQuery({ query: TeamInfoFromId, variables: { team_id: teamId } });
 				this.teamData = teamData?.community?.team || {};
+				setChallengeCookieData(this.cookieStore, teamId, this.teamData?.name);
```
Comment (dyersituations) [inline src/pages/Lend/LoanSearchPage.vue:244]: Following the general previous challenge behavior, shouldn't this be pushing the loan ID on add to basket?

File: src/pages/Lend/LoanSearchPage.vue [hunk 2]
```diff
@@ -241,7 +241,14 @@ export default {
 				this.challengeData = goalsData?.goals?.values?.[0] || {};
 				const teamData = this.apollo.readQuery({ query: TeamInfoFromId, variables: { team_id: teamId } });
 				this.teamData = teamData?.community?.team || {};
-				setChallengeCookieData(this.cookieStore, teamId, this.teamData?.name);
+
+				const loanIds = this.challengeData?.targets?.values?.map(target => target?.loanId) ?? [];
```
Comment (dyersituations) [inline src/pages/Lend/LoanSearchPage.vue:245]: The challenge doesn't just apply to the recommended loans. It applies to any loan added from `/lend/filter`. So we need to add/update this cookie value whenever the user adds to basket from `/lend/filter` when the challenge is active on that page for the current user.

File: src/pages/Lend/LoanSearchPage.vue [hunk 3]
```diff
@@ -240,6 +254,12 @@ export default {
 				this.challengeData = goalsData?.goals?.values?.[0] || {};
 				const teamData = this.apollo.readQuery({ query: TeamInfoFromId, variables: { team_id: teamId } });
 				this.teamData = teamData?.community?.team || {};
+
+				const loanIds = this.challengeData?.targets?.values?.map(target => target?.loanId) ?? [];
+
+				loanIds.forEach(loanId => {
```
Comment (roger-in-kiva) [inline src/pages/Lend/LoanSearchPage.vue:260]: I think we just need add loan ids when some of them are added to the basket successfully...

Comment (dyersituations) [reply src/pages/Lend/LoanSearchPage.vue:260]: Exactly. I don't think the changes in this method are needed, just the changes to `addLoanToChallengeCookie`

File: src/util/teamChallengeUtils.js [hunk 4]
```diff
@@ -56,3 +56,19 @@ export const removeLoansFromChallengeCookie = (cookieStore, loanIds) => {
...
+ * Sets challenge cookie data.
+ * Used when challenge is active on lend/filter.
+ *
+ * @param cookieStore The object for affecting cookies
+ * @param {Object} challengeData Challenge information
+ * @param {number} challengeData.loanId The ID of the loan
+ * @param {number} challengeData.teamId The ID of the team
+ * @param {string} challengeData.teamName The name of the team
+ */
+export const setChallengeCookieData = (cookieStore, challengeData) => {
```
Comment (dyersituations) [inline src/util/teamChallengeUtils.js:70]: Add tests for this new method (testing file exists for this file)

**Review by dyersituations — APPROVED** [review]

**Review by roger-in-kiva — APPROVED** [review]

**Pull request merged**